src/main/java/entity/MoonPhase.java

```diff
@@ -33,18 +33,28 @@ public String getImgURL() {
         return imageURL;
     }
 
-    public boolean validLatitude() {
-        final float floatLatitude = parseFloat(latitude);
-        return floatLatitude >= -90 && floatLatitude <= 90;
+    public boolean validLatitudeFormat() {
+        final String latitudePattern = "[-+]?[0-9]*\\.?[0-9]+";
+        return latitude.matches(latitudePattern);
     }
 
-    public boolean validLongitude() {
-        final float floatLongitude = parseFloat(longitude);
-        return floatLongitude >= -180 && floatLongitude <= 180;
+    public boolean validLongitudeFormat() {
+        final String longitudePattern = "[-+]?[0-9]*\\.?[0-9]+";
+        return longitude.matches(longitudePattern);
     }
 
-    public boolean validDate() {
+    public boolean validDateFormat() {
```

Comment:
This date format validation allows for invalid dates such as 9999-99-99, consider using a ``SimpleDateFormat`` instance and attempting to parse the date in a try-catch block to validate the format

Reply:
I see. I'll merge it for now and make a pr later on that fixes it.